samples/cmake/ozz-animation/main.cpp

```diff
@@ -0,0 +1,232 @@
+int main() {
+#if defined(NXDK)
+  XVideoSetMode(640, 480, 32, REFRESH_DEFAULT);
+#endif
+
+  SDL_Init(SDL_INIT_VIDEO);
+  SDL_Window* window = NULL;
+  SDL_Renderer* renderer = NULL;
+
+#if defined(NXDK)
+  debugPrint("Starting\n");
+#endif
+  SDL_CreateWindowAndRenderer(640, 480, 0, &window, &renderer);
+#if defined(NXDK)
+  debugPrint("Got %p / %p\n", window, renderer);
+#endif
+
+  bool ok = OnInitialize();
+
+#if defined(NXDK)
+  debugPrint("Mainloop: %d\n", ok);
+#endif
+  bool done = false;  
+  while (!done) {
+    SDL_Event event;
+
+    // Get animation time.
+    Uint32 duration_ticks = (Uint32)(animation_.duration() * 1000.0f);
+    Uint32 playback_ticks = SDL_GetTicks() % duration_ticks;
+    printf("%d / %d\n", playback_ticks, duration_ticks);
+
+    // Samples optimized animation.
+    ozz::animation::SamplingJob sampling_job;
+    sampling_job.animation = &animation_;
+    sampling_job.cache = &cache_;
+    sampling_job.ratio = (float)playback_ticks / (float)duration_ticks;
+    sampling_job.output = make_span(locals_);
+    if (!sampling_job.Run()) {
+      return false;
+    }
+
+    // Get number of joints.
+    const int num_joints = skeleton_.num_joints();
+    if (!num_joints) {
+      return true;
+    }
+
+    // Compute model space bind pose.
+    ozz::animation::LocalToModelJob job;
+    job.input = sampling_job.output;
+    job.output = ozz::make_span(models_);
+    job.skeleton = &skeleton_;
+    if (!job.Run()) {
+      return false;
+    }
+
+    const ozz::span<ozz::math::Float4x4>& _matrices = job.output;
+
+    // Get access to parents.
+    const ozz::span<const int16_t>& parents = skeleton_.joint_parents();
+
+    // Clear image.
+    SDL_SetRenderDrawColor(renderer, 0x00, 0x00, 0x00, SDL_ALPHA_OPAQUE);
+    SDL_RenderClear(renderer);
+
+    // Draw each bone.
+    int instances = 0;
+    for (int i = 0; i < num_joints; ++i) {
+      // Root isn't rendered.
+      const int16_t parent_id = parents[i];
+      if (parent_id == ozz::animation::Skeleton::kNoParent) {
+        continue;
+      }
+
+      // Selects joint matrices.
+      const ozz::math::Float4x4& parent = _matrices[parent_id];
+      const ozz::math::Float4x4& current = _matrices[i];
+
+      // Get position of the joints
+      float s = 200.0f;
+      int ox = 320;
+      int oy = 420;
+      int ax = parent.cols[3].z * s + ox;
+      int ay = -parent.cols[3].y * s + oy;
+      int bx = current.cols[3].z * s + ox;
+      int by = -current.cols[3].y * s + oy;
+
+      // Draw a line between joints to represent bone
+      if (IsLeaf(skeleton_, i)) {
+        SDL_SetRenderDrawColor(renderer, 0xFF, 0x00, 0x00, SDL_ALPHA_OPAQUE);
+      } else {
+        SDL_SetRenderDrawColor(renderer, 0xFF, 0xFF, 0xFF, SDL_ALPHA_OPAQUE);
+      }
+      SDL_RenderDrawLine(renderer, ax, ay, bx, by);
+    }
+   
+    SDL_RenderPresent(renderer);
+
+    while (SDL_PollEvent(&event)) {
+      if (event.type == SDL_QUIT) {
+        done = true;
+      }
+    }
+
+  }
+}
```

Comment:
When I updated ozz earlier I had to fix a bunch of stuff in this sample. I wasn't sure where this code originated, so I just patched some spots.

The ozz related code should be taken from a sample so it matches upstream again.

I know that this sample worked in the past.
However, I did not try running this sample since updating ozz.